model_analyzer/result/constraint_manager.py

```diff
@@ -75,3 +81,36 @@ def satisfies_constraints(constraints, run_config_measurement):
                                 return False
 
         return True
+
+    @staticmethod
+    def constraint_failure_percentage(
+            constraints: List[Dict[str, Dict[str, int]]],
+            run_config_measurement: 'RunConfigMeasurement') -> float:
+        """
+        Additive percentage, for every measurement, in every model, of how much 
+        the RCM is failing the constraints by
+        
+        Returns
+        -------
+        Float
+        """
+        failure_percentage = 0
```

Comment:
what are the bounds on this? 0 to 100? 0 to 100 * num_constraints?

Reply:
Technically, It's 0 to infinity, but in practice it's:

max constraint: 0 to infinity
min constraint: 0 to 100

And then each constraint is additive, along with being per model

Reply:
Is it possible to bound to 100 so the scale is the same as the code that compares objectives?

Reply:
No, I don't think you want to have a bound. Especially, in the situation where you can have multiple models and or constraints.

Reply:
So we want to use the same code to take steps via objective differences or constraint differences. Ideally we get return values in the same range. It isn't the end of the world if we are unbound, because we will clamp the step size.

Still, we were able to bound objectives by doing (A-B)/(A+B).  I haven't thought about it too closely, but can we do the same for constraints? We are doing A-B/A or B-A/A.  Can we not have A+B on the bottom?  That math probably doesn't make any sense, but just wanted to throw it out there.

Reply:
There not really analogusWe're doing A-B/100

The difference here is that we are comparing A against C (the constraint) and B against C. There's no good/easy way to compare A & B directly due to the fact that multiple constraints can be in play (across multiple models).